<?xml version="1.0"?>
<robot name="bicycle_drive" xmlns:xacro="http://ros.org/wiki/xacro">

    <ros2_control name="RealRobot" type="system">
        <hardware>
            <plugin>diffdrive_arduino/DiffDriveArduinoHardware</plugin>
            <param name="front_wheel_name">left_wheel_joint</param>
            <param name="rear_wheel_name">right_wheel_joint</param>
            <param name="loop_rate">30</param>
            <param name="device">
                /dev/serial/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1.3:1.0-port0</param>
            <param name="baud_rate">57600</param>
            <param name="timeout_ms">1000</param>
            <param name="enc_counts_per_rev">3436</param>
        </hardware>

        <joint name="steering_joint">
            <command_interface name="position" />
            <state_interface name="position" />
            <state_interface name="velocity" />
        </joint>

        <joint name="traction_joint">
            <command_interface name="velocity" />
            <state_interface name="position" />
            <state_interface name="velocity" />
        </joint>

    </ros2_control>

</robot>